packages/core/test/lib/integrations/third-party-errors-filter.test.ts

```diff
@@ -0,0 +1,247 @@
+import type { Client, Event } from '@sentry/types';
+import { GLOBAL_OBJ, createStackParser, nodeStackLineParser } from '@sentry/utils';
+import { thirdPartyErrorFilterIntegration } from '../../../src/integrations/third-party-errors-filter';
+
+function clone<T>(data: T): T {
+  return JSON.parse(JSON.stringify(data));
+}
+
+const stack = new Error().stack || '';
+
+const eventWithThirdAndFirstPartyFrames: Event = {
+  exception: {
+    values: [
+      {
+        stacktrace: {
+          frames: [
+            {
+              colno: 1,
+              filename: __filename,
+              function: 'function',
+              lineno: 1,
+            },
+            {
+              colno: 2,
+              filename: 'other-file.js',
+              function: 'function',
+              lineno: 2,
+            },
+          ],
+        },
+        type: 'SyntaxError',
+        value: 'missing ( on line 10',
+      },
+    ],
+  },
+};
+
+const eventWithOnlyFirstPartyFrames: Event = {
+  exception: {
+    values: [
+      {
+        stacktrace: {
+          frames: [
+            {
+              colno: 1,
+              filename: __filename,
+              function: 'function',
+              lineno: 1,
+            },
+            {
+              colno: 2,
+              filename: __filename,
+              function: 'function',
+              lineno: 2,
+            },
+          ],
+        },
+        type: 'SyntaxError',
+        value: 'missing ( on line 10',
+      },
+    ],
+  },
+};
+
+const eventWithOnlyThirdPartyFrames: Event = {
+  exception: {
+    values: [
+      {
+        stacktrace: {
+          frames: [
+            {
+              colno: 1,
+              filename: 'other-file.js',
+              function: 'function',
+              lineno: 1,
+            },
+            {
+              colno: 2,
+              filename: 'other-file.js',
+              function: 'function',
+              lineno: 2,
+            },
+          ],
+        },
+        type: 'SyntaxError',
+        value: 'missing ( on line 10',
+      },
+    ],
+  },
+};
+
+// This only needs the stackParser
+const MOCK_CLIENT = {
+  getOptions: () => ({
+    stackParser: createStackParser(nodeStackLineParser()),
+  }),
+} as unknown as Client;
+
+describe('ThirdPartyErrorFilter', () => {
```

Comment:
l: these tests lend themselves well to parameterization

Reply:
True, but it's a bit annoying since we are checking for defined-ness and that is not so easy to do in parameterized tests.